sql/core/src/main/scala/org/apache/spark/sql/execution/ExistingRDD.scala

```diff
@@ -336,9 +336,11 @@ private[sql] object DataSourceScanExec {
   def create(
       output: Seq[Attribute],
       rdd: RDD[InternalRow],
-      relation: BaseRelation,
+      logicalRelation: LogicalRelation,
       metadata: Map[String, String] = Map.empty,
       metastoreTableIdentifier: Option[TableIdentifier] = None): DataSourceScanExec = {
+    val relation: BaseRelation = logicalRelation.relation
+
     val outputPartitioning = {
       val bucketSpec = relation match {
```

Comment:
yea it's suboptimal, we should think harder about when should we disable bucketing. But for this PR, we are trying to fix a bug, improving bucketing is another story, we can do it after this PR. What do you think?